Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -1082,6 +1085,77 @@ theorem mkOfAlgebra {R S S' : Type _} [CommRing R] [CommRing S] [CommRing S'] [A
 
 end Algebra
 
+variable [Module (Localization S) M'] [IsScalarTower R (Localization S) M']
```

Comment:
I'm not an expert on this corner of the library but I believe we prefer to prove theorems about the universal property of localizations using `Submonoid.LocalizationMap` rather than the particular model provided by `Localization`.

With that in mind, I believe a better setup would be as follows:
```lean
variable {A : Type _} [CommRing A] [Algebra R A] [Module A M'] [IsScalarTower R A M']

theorem isBaseChange (g : Submonoid.LocalizationMap S A) : IsBaseChange A f := sorry -- Your proof here
```
and this recovers your original statement as:
```lean
-- Just for illustration, this wouldn't actually go in the library
example [Module (Localization S) M'] [IsScalarTower R (Localization S) M'] :
    IsBaseChange (Localization S) f :=
  isBaseChange S f (Localization.monoidOf S)
```
I'll raise this on Zulip to see if I can attract some expert opinions.

Reply:
I haven't used `LocalizedModule` but certainly my experience in ring/field theory localizations would agree with Oliver's suggestion.

Reply:
Me too, though I also haven't used `LocalizedModule`, and I think `(g : Submonoid.LocalizationMap S A)` should be `[IsLocalization S A]` (and then `(Localization.monoidOf S)` can be removed).

Reply:
Thanks @101damnations this makes sense, we have a special typeclass `IsLocalization` expressing the fact that an `algebraMap` satisfies the hypothesis required to build a `Submonoid.LocalizationMap`. So the correct statement should be:
```lean
theorem isBaseChange {A : Type _} [CommRing A] [Algebra R A] [Module A M'] [IsScalarTower R A M']
    [IsLocalization S A] :
    IsBaseChange A f :=
  sorry -- Your proof here!

-- Just for illustration, this wouldn't actually go in the library
example [Module (Localization S) M'] [IsScalarTower R (Localization S) M'] :
    IsBaseChange (Localization S) f :=
  isBaseChange S f
```